manifests/master/simp_auth.pp

```diff
@@ -1,55 +1,64 @@
 #
-# @param legacy_mcollective_all
-#   Type:    Boolean
-#   Default: true
-#   If enabled, allow access to the mcollective cacerts from the legacy `pki` module from all hosts
-#
 # @param legacy_pki_keytabs_from_host
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to each host's own kerberos keytabs from the legacy location
 #
 # @param pki_cacerts_all
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to the cacerts from the `pki_files` module from all hosts
-#
-# @param pki_mcollective_all
-#   Type:    Boolean
-#   Default: true
-#   If enabled, allow access to the mcollective PKI from the `pki_files` module from all hosts
+# @param pki_cacerts_all_rule
+#   The regex rule to match requests against. The provided rule matched requests
+#   coming from the `files/keydist/cacerts` directory from the pki_files module
+# @param pki_cacerts_all_allow
+# @param pki_cacerts_all_deny
 #
 # @param keydist_from_host
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to each host's own certs from the `pki_files` module
+# @param keydist_from_host_rule
+#   The regex rule to match requests against. The provided rule matched requests
+#   coming from the `files/keydist` directory from the pki_files module
+# @param keydist_from_host_allow Rules that the puppetserver should allow
+#   @see https://puppet.com/docs/puppetserver/2.7/config_file_auth.html#rules
+# @param keydist_from_host_deny Rules that the puppetserver should deny
+#   @see https://puppet.com/docs/puppetserver/2.7/config_file_auth.html#rules
 #
 # @param krb5_keytabs_from_host
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to each host's own kerberos keytabs from the `pki_files` module
+# @param krb5_keytabs_from_host_rule
+#   The regex rule to match requests against. The provided rule matched requests
+#   coming from the `files/keytabs` directory from the krb5_files module
+# @param krb5_keytabs_from_host_allow Rules that the puppetserver should allow
+#   @see https://puppet.com/docs/puppetserver/2.7/config_file_auth.html#rules
+# @param krb5_keytabs_from_host_deny Rules that the puppetserver should deny
+#   @see https://puppet.com/docs/puppetserver/2.7/config_file_auth.html#rules
 #
 class pupmod::master::simp_auth (
   Simplib::ServerDistribution $server_distribution          = simplib::lookup('simp_options::puppet::server_distribution', { 'default_value' => 'PC1' } ),
   Stdlib::AbsolutePath        $auth_conf_path               = '/etc/puppetlabs/puppetserver/conf.d/auth.conf',
   Boolean                     $legacy_cacerts_all           = false,
-  Boolean                     $legacy_mcollective_all       = false,
   Boolean                     $legacy_pki_keytabs_from_host = false,
   Boolean                     $pki_cacerts_all              = true,
-  Boolean                     $pki_mcollective_all          = true,
```

Comment:
How are we removing these two parameters and the associated `puppet_authorization::rule` declarations below without SemVer-bumping the major version?

Reply:
I'm not sure 100% how to handle it, but I think I'd prefer to gut the functionality and deprecate the params than do a version bump

Reply:
I'm putting them back, but the should be removed in https://simp-project.atlassian.net/browse/SIMP-4548